test/integration/test_estimator_v2.py

```diff
@@ -50,20 +53,78 @@ def test_estimator_v2_session(self, service):
-            self.assertIsInstance(result3, PrimitiveResult)
-            self.assertIsInstance(result3[2], PubResult)
-            self.assertIsInstance(result3[2].data, DataBin)
-            self.assertTrue(result3[2].metadata)
+            self._verify_result_type(result3, num_pubs=3, shapes=[(1,), (1,), (1,)])
+
+    @run_integration_test
+    def test_estimator_v2_options(self, service):
+        """Test V2 Estimator with different options."""
+        backend = service.backend(self.backend)
+        pass_mgr = generate_preset_pass_manager(backend=backend, optimization_level=1)
+        circuit = pass_mgr.run(IQP([[6, 5, 3], [5, 4, 5], [3, 5, 1]]))
+        observables = SparsePauliOp("X" * circuit.num_qubits).apply_layout(circuit.layout)
+
+        estimator = EstimatorV2(backend=backend)
+        estimator.options.resilience_level = 1
+        estimator.options.optimization_level = 1
+        # estimator.options.dynamical_decoupling = "XX"  # TODO: Re-enable when fixed
+        estimator.options.seed_estimator = 42
+        estimator.options.resilience.measure_noise_mitigation = True
+        estimator.options.resilience.zne_mitigation = True
+        estimator.options.resilience.zne_noise_factors = [3, 5]
+        estimator.options.resilience.zne_extrapolator = "linear"
+        estimator.options.resilience.zne_stderr_threshold = 0.25
+        estimator.options.resilience.pec_mitigation = False
+        estimator.options.execution.shots = 4000
+        estimator.options.execution.samples = 4
+        estimator.options.execution.shots_per_sample = 1000
+        estimator.options.execution.init_qubits = True
+        estimator.options.execution.interleave_samples = True
+        estimator.options.twirling.gates = True
+        estimator.options.twirling.measure = True
+        estimator.options.twirling.strategy = "active"
+
+        job = estimator.run([(circuit, observables)])
+        result = job.result()
+        self._verify_result_type(result, num_pubs=1, shapes=[(1,)])
+        self.assertEqual(result[0].metadata["shots"], 4000)
+        for res_key in ["twirled_readout_errors", "zne_noise_factors"]:
+            self.assertIn(res_key, result[0].metadata["resilience"])
+
+    @run_integration_test
+    def test_pec(self, service):
+        """Test running with PEC."""
+        backend = service.backend(self.backend)
+        pass_mgr = generate_preset_pass_manager(backend=backend, optimization_level=1)
+        circuit = pass_mgr.run(IQP([[6, 5, 3], [5, 4, 5], [3, 5, 1]]))
+        observables = SparsePauliOp("X" * circuit.num_qubits).apply_layout(circuit.layout)
+
+        estimator = EstimatorV2(backend=backend)
+        estimator.options.resilience_level = 0
+        estimator.options.optimization_level = 0
+        estimator.options.resilience.pec_mitigation = True
+        estimator.options.resilience.pec_max_overhead = 200
+        estimator.options.simulator.set_backend(service.backends(simulator=False)[0])
```

Comment:
I believe the cloud account we use for testing doesn't have any real backends so this might fail